spec/cypress/support/on-rails.js

```diff
@@ -0,0 +1,54 @@
+// CypressOnRails: dont remove these command
+Cypress.Commands.add('appCommands', function (body) {
+  cy.log("APP: " + JSON.stringify(body))
+  return cy.request({
+    method: 'POST',
+    url: "/__cypress__/command",
+    body: JSON.stringify(body),
+    log: true,
+    failOnStatusCode: true
+  }).then((response) => {
+    return response.body
+  });
+});
+
+Cypress.Commands.add('app', function (name, command_options) {
+  return cy.appCommands({name: name, options: command_options}).then((body) => {
+    return body[0]
+  });
+});
+
+Cypress.Commands.add('appScenario', function (name, options = {}) {
+  return cy.app('scenarios/' + name, options)
+});
+
+Cypress.Commands.add('appEval', function (code) {
+  return cy.app('eval', code)
+});
+
+Cypress.Commands.add('appFactories', function (options) {
+  return cy.app('factory_bot', options)
+});
+
+Cypress.Commands.add('appFixtures', function (options) {
+  cy.app('activerecord_fixtures', options)
+});
+// CypressOnRails: end
+
+// The next is optional
+// beforeEach(() => {
+//  cy.app('clean') // have a look at cypress/app_commands/clean.rb
+// });
```

Comment:
Cleaning the state before every test - is a good idea

Reply:
I'd be wary of doing this as it seems a bit magic, and it's not inconceivable that we'd have tests where we _wouldn't_ want to clean the state as the state isn't used—e.g., testing logged out pages, or testing pages on production.

However, I left the commented out code so that we could have this discussion :)

Reply:
I'd be for cleaning before every test. 

If we want to test that something is displayed after the action, I think we should explicitly set it up in that test - maybe commonising it when needed.

Reply:
Have changed it to clean before every test!